cms-contentful/src/test/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsServiceTest.java

```diff
@@ -45,42 +51,146 @@ public void whenSearchedContentExists_thenReturnIt() throws Exception {
     public void whenLanguageIsNotSupported_thenReturnOptionalEmpty() throws Exception {
         final Locale deAt = Locale.forLanguageTag("de-AT");
         final List<Locale> supportedLocales = Collections.singletonList(deAt);
-        final Optional<String> content = contentfulCmsService.getLocalizedField(supportedLocales, mockCdaEntry, FIELD_NAME);
+        final Optional<String> content = contentfulCmsService.getLocalizedContent(supportedLocales,
+                mockCdaEntry, FIELD_NAME);
 
         assertThat(content).isEmpty();
     }
 
     @Test
     public void whenThereIsNoMatchingFieldName_thenReturnOptionalEmpty() throws Exception {
         final String notMatchingFieldName = "notMatchingFieldName";
-        final Optional<String> content = contentfulCmsService.getLocalizedField(SUPPORTED_LOCALES, mockCdaEntry, notMatchingFieldName);
+        final Optional<String> content = contentfulCmsService.getLocalizedContent(SUPPORTED_LOCALES,
+                mockCdaEntry, notMatchingFieldName);
 
         assertThat(content).isEmpty();
     }
 
     @Test
     public void whenEntryDoesNotHaveRequiredField_thenReturnOptionalEmpty() throws Exception {
         final CDAEntry mockEntry = mockEntry(ENTRY_TYPE, ENTRY_KEY, "", CONTENT_VALUE);
-        final Optional<String> content = contentfulCmsService.getLocalizedField(SUPPORTED_LOCALES, mockEntry, FIELD_NAME);
+        final Optional<String> content = contentfulCmsService.getLocalizedContent(SUPPORTED_LOCALES,
+                mockEntry, FIELD_NAME);
 
         assertThat(content).isEmpty();
     }
 
-    private CDAEntry mockEntry(String entryType, String entryKey, String fieldName, String localizedFieldContent) {
-        final String entryTitle = "name";
+    @Test
+    public void whenEntryFieldTypeIsText_thenReturnOptionalString() throws Exception {
+        final CDAEntry mockCdaEntry = mockEntry(ENTRY_TYPE, ENTRY_KEY, FIELD_NAME, CONTENT_VALUE, TEXT);
+        final Optional<String> content = contentfulCmsService.getLocalizedContent(SUPPORTED_LOCALES,
+                mockCdaEntry, FIELD_NAME);
+
+        assertThat(content).contains(CONTENT_VALUE);
+    }
+
+    @Test
+    public void whenEntryFieldTypeIsDate_thenReturnOptionalString() throws Exception {
+        final String localizedFieldContent = "2015-11-06T09:45:27";
```

Comment:
Yes, it's fine that it's string, since it's returning "content", not a particular thing. And content is a string for sure, which can contain dates in some format, whole blocks of text and even HTML inside.